components/scream/src/physics/rrtmgp/share/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# Need to add Fortran as a language, since that is what the shr source is in
+enable_language(Fortran)
+
+# Set variables needed for processing genf90 templates
+set(CIMEROOT ${SCREAM_BASE_DIR}/../../cime)
+list(APPEND CMAKE_MODULE_PATH ${CIMEROOT}/src/CMake)
+set(GENF90 ${CIMEROOT}/src/externals/genf90/genf90.pl)
+set(ENABLE_GENF90 True)
+include(genf90_utils)
+include(Sourcelist_utils)
+
+# NOTE: I should be able to do this:
+#    # Note that this sets the share_sources variable to the list of source files we need to build
+#    add_subdirectory(${SCREAM_BASE_DIR}/../../share/util ${CMAKE_CURRENT_BINARY_DIR}/share_util_build)
+# However, that did not seem to add the fortran sources derived from the genf90 templates to the targets.
+# Instead, try explicitly adding sources and process the genf90 templates ourselves.
+# TODO: Figure out why the CMakeLists.txt and the cmake macros there do not handle this correctly.
+# Note also that our E3SM cmake build does not use the CMakeLists.txt in the share/util directory either
+# apparently, opting to do this manually as well.
+# GENF90_SOURCE lists source files we will need to run through the genf90 perl script
+set (GENF90_SOURCE 
+    ${SCREAM_BASE_DIR}/../../share/util/shr_infnan_mod.F90.in
+    ${SCREAM_BASE_DIR}/../../share/util/shr_assert_mod.F90.in
+)
+# FORTRAN_SOURCE lists the source files we want to build that do NOT need to be run through the genf90
+# perl script. We will append to this list below with our processed genf90 files.
+set (FORTRAN_SOURCE
+    shr_orb_mod_c.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_orb_mod.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_kind_mod.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_sys_mod.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_const_mod.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_log_mod.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_abort_mod.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_mpi_mod.F90
+    ${SCREAM_BASE_DIR}/../../share/util/shr_strconvert_mod.F90
+)
+# Process genf90 template files. This adds a custom command (and hence target) for each f90 source
+# that needs to be built from the genf90 template files listed in GENF90_SOURCE.
+foreach (SRC_FILE ${GENF90_SOURCE})
```

Comment:
I agree, but I don’t know how to avoid this here. I’m only using this because that’s how these sources are compiled in EAM’s cmake, and the implementation is taken directly from (this is compiling such “templated” F90 sources). If you have a better idea that does not entail us completely rewriting all of the shr modules in this PR, I’d be happy to implement it.

Reply:
Oh, sorry, I wasn't trying to say "do it another way". It was just a stand-alone rant. :)

However, in a full CIME build, I wonder if these modules may conflict with shared libs built by E3SM. I am mildly positive CIME will build this stuff at some point, and then we may have trouble at link time. Perhaps we should check if this is a CIME build, and, if not, proceed to build these files?